### PR TITLE
Support for additional P/Invoke libraries and additional native compile items

### PR DESCRIPTION
**Pull request by jeromelaban**

### CODE REVIEW

File: src/Uno.Wasm.Packager/packager.cs [hunk 1]
```diff
@@ -1141,6 +1144,16 @@ class WasmOptions {
...
 			ninja.WriteLine ($"  flags = {driver_cflags} -DDRIVER_GEN=1 -I{runtime_dir}/include/mono-2.0");
 
+			foreach (var nativeCompile in native_compile)
+			{
+				var fileName = Path.GetFileName(nativeCompile);
+				var fileNameWithoutExt = Path.GetFileNameWithoutExtension(nativeCompile);
+
+				ninja.WriteLine($"build $builddir/{fileName}: cpifdiff {source_file}");
+				ninja.WriteLine($"build $builddir/{fileNameWithoutExt}.o: emcc {nativeCompile} | $emsdk_env {driver_deps}");
+				ninja.WriteLine($"  flags = {driver_cflags} -DDRIVER_GEN=1 -I{runtime_dir}/include/mono-2.0");
```
Comment (ebariche) [inline src/Uno.Wasm.Packager/packager.cs:1154]: Do we want to support additionnal includes here ?

Comment (ebariche) [reply src/Uno.Wasm.Packager/packager.cs:1154]: What about defines ?

**Comment by ebariche** [comment]
Some thoughts:

- Do we want to support custom js support libs via --js-library ? In case the native code calls into js ? That way we could use TS.
- If those additionnal modules need initialization where would that occur ? What about passing in some args ?

File: src/Uno.Wasm.Packager/packager.cs (hunk 1 above)
Comment (jeromelaban) [reply src/Uno.Wasm.Packager/packager.cs:1154]: Eventually, but this feature is more about ad-hoc dev, not so much about a full-fledged C/C++ compiler toolchain :) I'll update the docs to mention this.

**Comment by jeromelaban** [comment]
> Some thoughts:
> 
> * Do we want to support custom js support libs via --js-library ? In case the native code calls into js ? That way we could use TS.
> * If those additionnal modules need initialization where would that occur ? What about passing in some args ?

The scenario is really about C# <-> C/C++ interop, nothing additional at this point.

**Pull request merged**